SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityTaskReconciliation.java

```diff
@@ -94,6 +95,17 @@ boolean isReconciliationRunning() {
   }
 
   public ReconciliationState startReconciliation() {
+    if (
+      (float) taskManager.getNumScheduledTasks() /
+      configuration.getOfferCacheSize() >=
+      NEARLY_FULL
+    ) {
+      LOG.info("Queue is nearly full, not starting reconciliation");
+      return isRunningReconciliation.get()
```

Comment:
based on the issue we saw, we were already in a state of reconciliation when we entered the endless reconciliation attempt loop, so I wanted to base the return off of that possibility

Reply:
I think we also check this a few lines down

Reply:
but this is short circuiting all of that logic, should I not short circuit?

Reply:
I think the `if (!isRunningReconciliation.compareAndSet(false, true)) {` should accomplish the same short circuit. Which still makes me wonder about how we got multiple of those loops in there 🤔

Reply:
hmm you're right, that is odd. maybe there is a different bug?